mira/metamodel/templates.py

```diff
@@ -66,6 +168,58 @@ def from_json(cls, data) -> "Template":
         stmt_cls = getattr(sys.modules[__name__], template_type)
         return stmt_cls(**data)
 
+    def is_equal_to(self, other: "Template", with_context: bool = False) -> bool:
+        if not isinstance(other, Template):
+            return False
+        return templates_equal(self, other, with_context)
+
+    def refinement_of(
+        self,
+        other: "Template",
+        refinement_func: Callable[[str, str], bool],
+        with_context: bool = False,
+    ) -> bool:
+        """Check if this template is a more detailed version of another"""
+        if not isinstance(other, Template):
+            return False
+
+        if self.type != other.type:
+            return False
+
+        for field_name in self.__fields__:
```

Comment:
can you operate over `self.dict().items()` instead to avoid the introspection?

Reply:
To get the keys I suppose we could, but `getattr` is still needed to get the `Concept` instance rather than the dict version of it (so we can use `refinement_of`)

Reply:
Resolved in https://github.com/indralab/mira/pull/19/commits/a2e3870fab6b8d7bc0acce635491b282d1535dc1